### PR TITLE
[nrf noup] Add MPC and SPC error reporting for nrf54l

### PR DESCRIPTION
**Pull request by vili-nordic**
Adds handling of MPC and SPC errors.

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

**Review by mswarowsky — APPROVED** [review]
This is gonna be so nice to have the address in the report.

Could you add maybe 
`fixup! [nrf noup] platform: nordic_nrf: Add support for 54l`
as first line of the commit description so this will get squashed no the next upmerge

File: platform/ext/target/nordic_nrf/common/core/nrf_exception_info.c [hunk 1]
```diff
@@ -11,34 +11,60 @@
 
 static struct nrf_exception_info nrf_exc_info;
 
-static void spu_dump_context(struct nrf_exception_info *ctx)
+static void dump_exception_info(struct nrf_exception_info *ctx)
 {
     SPMLOG_ERRMSG("Platform Exception: SPU Fault\r\n");
```
Comment (Vge0rge) [inline platform/ext/target/nordic_nrf/common/core/nrf_exception_info.c:16]: Very nit but this should write something like more generic now that MPC maybe caused the fault?

File: platform/ext/target/nordic_nrf/common/core/nrf_exception_info.c [hunk 2]
```diff
@@ -11,34 +11,60 @@
...
+        SPMLOG_DBGMSGVAL("    Execute:      ", (ctx->memaccerr.info >> 14) & 1);
+        SPMLOG_DBGMSGVAL("    Secure:       ", (ctx->memaccerr.info >> 15) & 1);
+        SPMLOG_DBGMSGVAL("    Error source: ", (ctx->memaccerr.info >> 16) & 1);
+    }
+#endif
 }
 
 void nrf_exception_info_store_context(void)
 {
     nrf_exc_info.events = spu_events_get();
```
Comment (Vge0rge) [inline platform/ext/target/nordic_nrf/common/core/nrf_exception_info.c:50]: Does that not need to be inside the ifdef?

Comment (vili-nordic) [reply platform/ext/target/nordic_nrf/common/core/nrf_exception_info.c:50]: No, this is platform independent, the code in the ifdefs are to get extra information on supported platforms.

**Pull request merged**